main.js

```diff
@@ -0,0 +1,190 @@
+    displayRecipes(recipes);
+  } catch (error) {
+    console.error('Error fetching recipes:', error);
+  }
+}
+
+function displayRecipe(recipe) {
+  const container = document.getElementById("recipe-container");
+  const recipeElement = document.createElement("div");
+  recipeElement.className = "recipe";
+
+  const title = document.createElement("h2");
+  title.textContent = recipe.name;
+  recipeElement.appendChild(title);
+
+  const description = document.createElement("p");
+  description.textContent = recipe.description;
+  recipeElement.appendChild(description);
+
+  const img = document.createElement("img");
+  img.src = recipe.pictureUrl;
+  img.alt = recipe.name;
+  recipeElement.appendChild(img);
+
+  const ingredientsTitle = document.createElement("h3");
+  ingredientsTitle.textContent = "Ingredients:";
+  recipeElement.appendChild(ingredientsTitle);
+
+  const ingredientsList = document.createElement("ul");
+  recipe.ingredients.forEach((ingredient) => {
+    const listItem = document.createElement("li");
+    listItem.textContent = ingredient;
+    ingredientsList.appendChild(listItem);
+  });
+  recipeElement.appendChild(ingredientsList);
+
+  container.appendChild(recipeElement);
+}
+
+function displayRecipes(recipeArray) {
+  const recipeGrid = document.getElementById("recipe-grid");
+  recipeGrid.innerHTML = "";
+
+  recipeArray.forEach((recipe) => {
+    const recipeCard = document.createElement("div");
+    recipeCard.className = "recipe-card";
+    recipeCard.innerHTML = `
+      <h2>${recipe.name}</h2>
+      <p>Ingredients:</p>
+      <ul>
+        ${recipe.ingredients.map((ingredient) => `<li>${ingredient}</li>`).join("")}
+      </ul>
+    `;
+    recipeGrid.appendChild(recipeCard);
+  });
+}
+
+async function addNewRecipe(newRecipe) {
+  try {
+    const response = await fetch('https://raw.githubusercontent.com/antuanettaV/antuanettaV.github.io/main/test.json', { 
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(newRecipe),
+    });
+    if (!response.ok) throw new Error('Failed to add new recipe');
+    const addedRecipe = await response.json();
+    recipes.push(addedRecipe); 
+    displayRecipes(recipes); 
+  } catch (error) {
+    console.error('Error adding new recipe:', error);
+  }
+}
+
+document.getElementById("recipe-form").addEventListener("submit", function (event) {
+  event.preventDefault();
+  const title = document.getElementById("title").value;
+  const description = document.getElementById("description").value;
+  const pictureUrl = document.getElementById("pictureUrl").value;
+  const ingredients = [];
+
+  for (let i = 1; i <= ingredientCount; i++) {
+    const ingredient = document.getElementById(`ingredient${i}`).value;
+    if (ingredient) ingredients.push(ingredient);
+  }
+
+  const newRecipe = {
+    id: Date.now(),
+    name: title,
+    description,
+    ingredients,
+    pictureUrl
+  };
+
+  addNewRecipe(newRecipe);
```

Comment:
`addNewRecipe` is an `async` function so you should add `await` before calling it to make sure that the API call is performed before continuing with the rest of code.

```suggestion
await addNewRecipe(newRecipe);
```